com.unity.render-pipelines.universal/Runtime/2D/Light2D.cs

```diff
@@ -218,35 +222,26 @@ internal int GetTopMostLitLayer()
 
         internal void UpdateMesh(bool forceUpdate)
         {
-            var shapePathHash = LightUtility.GetShapePathHash(shapePath);
-            var fallOffSizeChanged = LightUtility.CheckForChange(m_ShapeLightFalloffSize, ref m_PreviousShapeLightFalloffSize);
-            var parametricRadiusChanged = LightUtility.CheckForChange(m_ShapeLightParametricRadius, ref m_PreviousShapeLightParametricRadius);
-            var parametricSidesChanged = LightUtility.CheckForChange(m_ShapeLightParametricSides, ref m_PreviousShapeLightParametricSides);
-            var parametricAngleOffsetChanged = LightUtility.CheckForChange(m_ShapeLightParametricAngleOffset, ref m_PreviousShapeLightParametricAngleOffset);
-            var spriteInstanceChanged = LightUtility.CheckForChange(lightCookieSpriteInstanceID, ref m_PreviousLightCookieSprite);
-            var shapePathHashChanged = LightUtility.CheckForChange(shapePathHash, ref m_PreviousShapePathHash);
-            var lightTypeChanged = LightUtility.CheckForChange(m_LightType, ref m_PreviousLightType);
-            var hashChanged = fallOffSizeChanged || parametricRadiusChanged || parametricSidesChanged ||
-                             parametricAngleOffsetChanged || spriteInstanceChanged || shapePathHashChanged || lightTypeChanged;
-            // Mesh Rebuilding
-            if (hashChanged && forceUpdate)
+            int hashCode = GetHashCode();
+            if (forceUpdate && hashCode != m_HashCode)
```

Comment:
forceUpdate is false when called onAwake when mesh is cached. On this specific case, we don't need to call Mesh Generation. Hence its an and.

Otherwise its always true.